### PR TITLE
exo-bench (Benchmark model pp & tg speed)

### PR DESCRIPTION
**Pull request by rltakashige**
## Motivation

This PR implements benchmarking in the style of llama-bench. The main difficulty here is the fact that exo is not a library - it exposes an endpoint. This means that benchmarking numbers will be inaccurate if the API is measured.

The solution assumes nodes are set up with uv run exo (or via the app), and then hits the new endpoint /bench/chat/completions to retrieve generation statistics directly from mlx_lm. 

This will allow us to release benchmarks for models and perform regression tests.

TODO: Performance benchmarking.

## Changes

- Adds /bench/chat/completions endpoint
- Adds BenchChatCompletion/Response
- Adds a logits processor to prevent response from ending early
- Adds a "Prompt Sizer" which downloads the tokenizer and dynamically adjusts the prompt of "a" to fit the desired prompt size.
- Reduce prefill step size to 2048 for now (in future, dynamically adjust this value)

### Manual Testing

Benchmarked Llama, Qwen, DeepSeek and Kimi models. Will require several fixes to run consistently on all configurations (to be done in the future). 
Manually tested the normal API to verify chat requests complete as expected.

### Automated Testing

Not really possible. Type checker passes.

### CODE REVIEW

File: src/exo/worker/engines/mlx/generator/generate.py [hunk 1]
```diff
@@ -72,25 +77,47 @@ def warmup_inference(
...
+    return eos
+
+
 def mlx_generate(
     model: Model,
     tokenizer: TokenizerWrapper,
     sampler: Callable[[mx.array], mx.array],
     task: ChatCompletionTaskParams,
 ) -> Generator[GenerationResponse]:
+    mx.reset_peak_memory()
```
Comment (JakeHillion) [inline src/exo/worker/engines/mlx/generator/generate.py:122]: Why?

File: src/exo/worker/engines/mlx/generator/generate.py [hunk 2]
```diff
@@ -72,25 +77,47 @@ def warmup_inference(
...
         kv_group_size=KV_GROUP_SIZE,
         kv_bits=KV_BITS,
     ):
         logger.info("Generated warmup token: " + str(_r.text))
         tokens_generated += 1
 
     logger.info("Generated ALL warmup tokens")
-    mx_barrier()
+
+    # TODO: Do we want an mx_barrier?
```
Comment (JakeHillion) [inline src/exo/worker/engines/mlx/generator/generate.py:90]: Might be worth adding this back for now, if it's not essential to this PR. There's an open discussion about whether we need more of these barriers.

**Review by JakeHillion — APPROVED** [review]
Couple of comments of bits of code that likely need commenting, but this looks great.
